gammapy/estimators/map/excess.py

```diff
@@ -195,42 +195,63 @@ def run(self, dataset):
         result = self.estimate_excess_map(resampled_dataset)
         return result
 
-    def estimate_excess_map(self, dataset):
-        """Estimate excess and ts maps for single dataset.
-
-        If exposure is defined, a flux map is also computed.
+    def estimate_kernel(self, dataset):
+        """Get the convolution kernel for the input dataset.
 
         Parameters
         ----------
-        dataset : `MapDataset`
-            Map dataset
-        """
+        dataset : `~gammapy.datasets.MapDataset`
+            Input dataset.
 
+        Returns
+        -------
+        kernel : `~astropy.convolution.Tophat2DKernel`
+            Kernel
+        """
         pixel_size = np.mean(np.abs(dataset.counts.geom.wcs.wcs.cdelt))
         size = self.correlation_radius.deg / pixel_size
         kernel = Tophat2DKernel(size)
+        return kernel
 
-        geom = dataset.counts.geom
+    def get_mask(self, dataset):
+        """Get mask used by the estimator.
 
+        Parameters
+        ----------
+        dataset : `~gammapy.datasets.MapDataset`
+            Input dataset.
+
+        Returns
+        -------
+        mask : `Map`
+            Mask map
+        """
         if dataset.mask_fit:
             mask = dataset.mask
         elif dataset.mask_safe:
             mask = dataset.mask_safe
         else:
-            mask = Map.from_geom(geom, data=True, dtype=bool)
+            mask = Map.from_geom(dataset.counts.geom, data=True, dtype=bool)
+        return mask
 
-        counts_stat = convolved_map_dataset_counts_statistics(
-            dataset, kernel, mask, self.correlate_off
-        )
+    def estimate_reco_exposure(self, dataset, kernel, mask):
```

Comment:
Rename to `estimate_exposure_reco_energy` for consistency with the utility method.